components/layout_2020/positioned.rs

```diff
@@ -775,24 +772,52 @@ impl<'a> AbsoluteAxisSolver<'a> {
     ///
     /// In the replaced case, `size` is never `Auto`.
     fn solve(&self, get_content_size: Option<impl FnOnce() -> ContentSizes>) -> AxisResult {
-        let solve_for_anchor = |anchor: Anchor| {
-            let margin_start = self.computed_margin_start.auto_is(Au::zero);
-            let margin_end = self.computed_margin_end.auto_is(Au::zero);
-            let size = self
-                .computed_size
-                .non_auto()
-                .or_else(|| {
-                    let content_size = get_content_size?();
-                    let available_size = self.containing_size -
-                        anchor.inset() -
-                        self.padding_border_sum -
-                        margin_start -
-                        margin_end;
-                    Some(content_size.shrink_to_fit(available_size))
+        let mut get_content_size = get_content_size.map(|get_content_size| {
+            let content_size = LazyCell::new(get_content_size);
+            move || *content_size
+        });
```

Comment:
Is the idea here that you'd like to cache the results of the call to `get_content_size` so that if the return value is called more than once the inner `get_content_size` is only called once?

Reply:
Yes, it can be called more than once if various of `min-width`, `max-width` and `width` (or height) are intrinsic.

Reply:
Being called once is ensured by `FnOnce`, but I have added this comment:

```rs
// The provided `get_content_size` is a FnOnce but we may need its result multiple times.
// A LazyCell will only invoke it once if needed, and then reuse the result.
```